apis/kueue/v1alpha1/workload_webhook.go

```diff
@@ -58,15 +62,41 @@ var _ webhook.Validator = &Workload{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Workload) ValidateCreate() error {
-	return nil
+	return validateWorkload(r).ToAggregate()
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Workload) ValidateUpdate(old runtime.Object) error {
-	return nil
+	return validateWorkload(r).ToAggregate()
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
 func (r *Workload) ValidateDelete() error {
 	return nil
 }
+
+func validateWorkload(obj *Workload) field.ErrorList {
+	var allErrs field.ErrorList
+	specField := field.NewPath("spec")
+	for i, podSet := range obj.Spec.PodSets {
+		if podSet.Count <= 0 {
+			allErrs = append(allErrs, field.Invalid(
+				specField.Child("podSets").Index(i).Child("count"),
+				podSet.Count,
+				"count must be greater than 0"),
+			)
+		}
+	}
+
+	if len(obj.Spec.PriorityClassName) > 0 {
+		msgs := validation.IsDNS1123Subdomain(obj.Spec.PriorityClassName)
```

Comment:
replace with something like below as we may have multi different type errors, not different msgs for one error.
```
		fldPath := specField.Child("priorityClassName")
		for _, msg := range validation.IsDNS1123Subdomain(obj.Spec.PriorityClassName) {
			allErrs = append(allErrs, field.Invalid(fldPath, obj.Spec.PriorityClassName, msg))
		}
```

Reply:
Thanks! I have updated the code.